edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -12,41 +12,52 @@ public class CertificateRenewal : IDisposable
         static readonly TimeSpan TimeBuffer = TimeSpan.FromMinutes(5);
 
         readonly TimeSpan maxRenewAfter;
+        readonly TimeSpan maxCheckCertExpiryAfter;
         readonly EdgeHubCertificates certificates;
         readonly ILogger logger;
         readonly Timer timer;
         readonly CancellationTokenSource cts;
 
-        public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger, TimeSpan maxRenewAfter)
+        public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger, TimeSpan maxRenewAfter, TimeSpan maxCheckCertExpiryAfter)
```

Comment:
I feel this logic very complicated, why doesn't it just setup a timer with a short time (e.g. 1 min or 5 min or configured) and does the certificate expiration check with the actual time? Is it expensive to check often? Or we don't want to pollute the logs?
What I am saying is to
- delete this thing below
- delete the "callback"
- setup a timer to PeriodicCallback

Wouldn't that work?

Reply:
yes, that is basically what the new logic does. 
Not deleting existing logic to avoid any change in behavior for existing customers.